test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -4202,6 +4210,365 @@ defmodule Trento.SapSystemTest do
     end
   end
 
+  describe "instance marked absent/present" do
+    test "should mark as absent a previously registered database instance" do
+      sap_system_id = Faker.UUID.v4()
+      sid = fake_sid()
+      host_id = Faker.UUID.v4()
+      absent_db_instance_number = "01"
+      present_db_instance_number = "02"
+      absent_db_absent_at = DateTime.utc_now()
+
+      initial_events = [
+        build(
+          :database_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        ),
+        build(
+          :database_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          host_id: host_id,
+          instance_number: absent_db_instance_number,
+          system_replication: nil,
+          system_replication_status: nil
+        ),
+        build(
+          :database_instance_marked_absent_event,
+          sap_system_id: sap_system_id,
+          host_id: host_id,
+          instance_number: absent_db_instance_number,
+          absent_at: absent_db_absent_at
+        ),
+        build(
+          :database_instance_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid,
+          host_id: host_id,
+          instance_number: present_db_instance_number,
+          system_replication: nil,
+          system_replication_status: nil
+        )
+      ]
+
+      absent_at = DateTime.utc_now()
+
+      assert_events_and_state(
+        initial_events,
+        [
+          %MarkDatabaseInstanceAbsent{
+            instance_number: absent_db_instance_number,
+            host_id: host_id,
+            sap_system_id: sap_system_id,
+            absent_at: absent_at
+          },
+          %MarkDatabaseInstanceAbsent{
+            instance_number: present_db_instance_number,
+            host_id: host_id,
+            sap_system_id: sap_system_id,
+            absent_at: absent_at
+          }
+        ],
+        [
+          %DatabaseInstanceMarkedAbsent{
+            instance_number: present_db_instance_number,
+            host_id: host_id,
+            sap_system_id: sap_system_id,
+            absent_at: absent_at
+          }
+        ],
+        fn state ->
+          assert %SapSystem{
+                   database: %SapSystem.Database{
```

Comment:
`SapSystem.Database` to `Database` (already aliased in top).
The same for `Instance` and `Application` in all tests